src/sparkle/writer/kafka_writer.py

```diff
@@ -109,3 +109,91 @@ def write(self, df: DataFrame) -> None:
         )
+
+
+class KafkaBatchPublisher(Writer):
+    """KafkaBatchublisher class for writing DataFrames in batch to Kafka.
+
+    Inherits from the Writer abstract base class and implements the write
+    method for writing data to Kafka topics.
+
+    Args:
+        kafka_options (dict[str, Any]): Kafka connection options.
+        kafka_topic (str): Kafka topic to which data will be written.
+        unique_identifier_column_name (str): Column name used as the Kafka key.
+        spark (SparkSession): Spark session instance to use.
+    """
+
+    def __init__(
+        self,
+        kafka_options: dict[str, Any],
+        kafka_topic: str,
+        unique_identifier_column_name: str,
+        spark: SparkSession,
+    ) -> None:
+        """Initialize the KafkaBatchPublisher object.
+
+        Args:
+            kafka_options (dict[str, Any]): Kafka options for the connection.
+            kafka_topic (str): The target Kafka topic for writing data.
+            unique_identifier_column_name (str): Column name to be used as Kafka key.
+            spark (SparkSession): The Spark session to be used for writing data.
+        """
+        super().__init__(spark)
+        self.kafka_options = kafka_options
+        self.kafka_topic = kafka_topic
+        self.unique_identifier_column_name = unique_identifier_column_name
+
+    @classmethod
+    def with_config(cls, config: Config, spark: SparkSession, **kwargs: Any) -> "KafkaBatchPublisher":
+        """Create a KafkaBatchPublisher object with a configuration.
+
+        Args:
+            config (Config): Configuration object containing settings for the writer.
+            spark (SparkSession): The Spark session to be used for writing data.
+            **kwargs (Any): Additional keyword arguments.
+
+        Returns:
+            KafkaBatchPublisher: An instance configured with the provided settings.
+        """
+        if not config.kafka_output:
+            raise ValueError("Kafka output configuration is missing")
+
+        c = config.kafka_output
+
+        return cls(
+            kafka_options=c.kafka_config.spark_kafka_config,
+            kafka_topic=c.kafka_topic,
+            unique_identifier_column_name=c.unique_identifier_column_name,
+            spark=spark,
+        )
+
+    def write(self, df: DataFrame) -> None:
+        """Write DataFrame to Kafka by converting it to JSON using the configured primary key.
+
+        This method transforms the DataFrame using the unique identifier column name
+        and writes it to the configured Kafka topic.
+
+        Args:
+            df (DataFrame): The DataFrame to be written.
+
+        Raises:
+            KeyError: If the DataFrame does not have the required 'key' and 'value' columns.
+        """
+        # Convert the DataFrame to a Kafka-friendly format
+        kafka_df = to_kafka_dataframe(self.unique_identifier_column_name, df)
+
+        if "key" not in kafka_df.columns or "value" not in kafka_df.columns:
```

Comment:
```suggestion
        if set(kafka_df.columns) != {"key", "value"}
```

I think an approach like this would be safer, ensuring user is not putting any extra column, hoping it'd show up in the Kafka stream.

Reply:
Good idea, we need to change it also from the other writer cause I really just duplicated the class and copied most things.